format/FlightSql.proto

```diff
@@ -867,6 +867,167 @@ enum SqlSupportsConvert {
+   */
+  SEARCHABLE_NONE = 0;
+
+  /**
+   * Indicates that the column can be used in a WHERE clause if it is using a
+   * LIKE predicate.
+   */
+  SEARCHABLE_CHAR = 1;
+
+  /**
+   * Indicates that the column can be used in a WHERE clause using other predicates
+   * except for LIKE.
+   *
+   * - Allowed operators: comparison, quantified comparison, BETWEEN,
+   *                      DISTINCT, IN, MATCH, and UNIQUE.
+   */
+  SEARCHABLE_BASIC = 2;
+
+  /**
+   * Indicates that the column can be used in a WHERE clause using any operator.
+   */
+  SEARCHABLE_FULL = 3;
+}
+
+/*
+ * Represents a request to retrieve information about data type supported on a Flight SQL enabled backend.
+ * Used in the command member of FlightDescriptor for the following RPC calls:
+ *  - GetSchema: return the schema of the query.
+ *  - GetFlightInfo: execute the catalog metadata request.
+ *
+ * The returned schema will be:
+ * <
+ *   type_name: utf8 not null (The name of the data type, for example: VARCHAR, INTEGER, etc),
+ *   data_type: int not null (The SQL data type),
+ *   column_size: int (The maximum size supported by that column.
+ *                     In case of numeric types, this represents the maximum precision.
+ *                     In case of string types, this represents the character length.
+ *                     In case of datetime data types, this represents the length in characters of the string representation.
+ *                     NULL is returned for data types where column size is not applicable.),
+ *   literal_prefix: utf8 (Character or characters used to prefix a literal, NULL is returned for
+ *                         data types where a literal prefix is not applicable.),
+ *   literal_suffix: utf8 (Character or characters used to terminate a literal,
+ *                         NULL is returned for data types where a literal suffix is not applicable.),
+ *   create_params: list<utf8 not null>
+ *                        (A list of keywords corresponding to which parameters can be used when creating
+ *                         a column for that specific type.
+ *                         NULL is returned if there are no parameters for the data type definition.),
+ *   nullable: int not null (Shows if the data type accepts a NULL value. The possible values can be seen in the
+ *                           Nullable enum.),
+ *   case_sensitive: bool not null (Shows if a character data type is case-sensitive in collations and comparisons),
+ *   searchable: int not null (Shows how the data type is used in a WHERE clause. The possible values can be seen in the
+ *                             Searchable enum.),
+ *   unsigned_attribute: bool (Shows if the data type is unsigned. NULL is returned if the attribute is
+ *                             not applicable to the data type or the data type is not numeric.),
+ *   fixed_prec_scale: bool not null (Shows if the data type has predefined fixed precision and scale.),
```

Comment:
Some data sources have specific type that come with a fixed precision and scale values such as money data type. This will retrieve that information

Reply:
Er... what does "fixed" mean here exactly?

Reply:
That it does not change, it will be always the same

Reply:
Please be more precise. In which cases would the precision change?

Reply:
Here we are not talking about columns, but the possible types a database can have. And for some types, such as money, they have a precision and a scale predefined by the database, the user cannot alter it when create a table with a column of such type

Reply:
But there are some types where the precision would be given in a field here but still be able to change? Can you give an example?

Reply:
No, the precision and the scale does not change inside a column. All the values from the column have the same precision and scale.

This property "fixed_prec_scale" is from a data type, for example the data type **MONEY** has a fixed_prec_scale of true in the **SQL Server**. When declaring a column as MONEY, you do not set the precision and the scale.

In the Decimal Type, it has a fixed_prec_scale as false. So, when declaring it you can set the precision and the scale.

Reply:
> In the Decimal Type, it has a fixed_prec_scale as false. So, when declaring it you can set the precision and the scale.

Ok, but in this case, I suppose the precision and scale fields are empty?

Reply:
No, it would have the precision and scale determine by its type.

In the case of the Decimal Type for the SQL Server we used as example above, I would have this as response:

`
FIXED_PREC_SCALE = false
MINIMUM_SCALE = 0
MAXIMUM_SCALE =
   28 (SQL Server 7), *
   38 (SQL Server 2000 and SQL Server 2005) 
PRECISION =
   28 (SQL Server 7) *,
   38 (SQL Server 2000 and SQL Server 2005) *`

Reply:
There is also a slight difference in semantics when `fixed_prec_scale` is true.  `MONEY` can generally be thought of as a decimal with precision 19 and scale 4 but consider this example on SQL Server 2017...

```
CREATE TABLE sample (mon MONEY, deci DECIMAL(10, 4));
INSERT INTO sample (mon, deci) VALUES (1, 1);
SELECT mon / 100000, deci / 100000 FROM sample;
```

This will return:

| mon / 100000 | deci / 100000 |
| ------------- | ------------- |
| 0  | 0.00001  |

This is because (I believe) a data type with `fixed_prec_scale = false` is not allowed to be widened as a natural result of arithmetic operations.

Reply:
Ok, thanks for the detailed explanations @jcralmeida and @westonpace !